lib/associations/has-one.js

```diff
@@ -190,9 +190,7 @@ class HasOne extends Association {
    * @returns {Promise}
    */
   async set(sourceInstance, associatedInstance, options) {
-    options = Object.assign({}, options, {
-      scope: false
-    });
+    options = { ...options, scope: false };
```

Comment:
ideally parameters should be read-only, but that's another problem

Reply:
Correct, I've seen numerous fixes creeping in over time that end up cloning/deep cloning inputs, but this almost impossible to fix on a greater scale without static analysis.

Reply:
Totally agree